content/post/2022-04-28-flower/index.html

```diff
@@ -0,0 +1,501 @@
+---
+title: "echarts4r 画花"
+author: "袁凡"
+date: "2022-04-28"
+categories:
+  - R 语言
+tags:
+  - echarts4r
+  - Apache Echarts
+output:
+  blogdown::html_page:
+    toc: true
+link-citations: true
+forum_id: 423132
+slug: echarts4r-flower
+---
+
+<script src="{{< blogdown/postref >}}index_files/htmlwidgets/htmlwidgets.js"></script>
+<script src="{{< blogdown/postref >}}index_files/echarts4r/echarts-en.min.js"></script>
+<script src="{{< blogdown/postref >}}index_files/echarts4r/ecStat.min.js"></script>
+<script src="{{< blogdown/postref >}}index_files/echarts4r/dataTool.min.js"></script>
+<script src="{{< blogdown/postref >}}index_files/echarts4r-binding/echarts4r.js"></script>
```

Comment:
哦这样不行，如果定稿后在 PR 内删掉 `index_files` 文件夹，那么 PR 预览里也就相应删掉了。得用最后一次预览的永久地址：

```suggestion
<script src="https://6271e437dd6bd60009fb46fe--cosx.netlify.app/2022/04/echarts4r-flower/index_files/htmlwidgets/htmlwidgets.js"></script>
<script src="https://6271e437dd6bd60009fb46fe--cosx.netlify.app/2022/04/echarts4r-flower/index_files/echarts4r/echarts-en.min.js"></script>
<script src="https://6271e437dd6bd60009fb46fe--cosx.netlify.app/2022/04/echarts4r-flower/index_files/echarts4r/ecStat.min.js"></script>
<script src="https://6271e437dd6bd60009fb46fe--cosx.netlify.app/2022/04/echarts4r-flower/index_files/echarts4r/dataTool.min.js"></script>
<script src="https://6271e437dd6bd60009fb46fe--cosx.netlify.app/2022/04/echarts4r-flower/index_files/echarts4r-binding/echarts4r.js"></script>
```

我刚在主分支里做了个更改：ac0e5e4bdcc08132adb91d67e2ec7a73b86a9daa 以后这个永久地址就可以很容易从预览页面的 HTML 源码里找到了。

Reply:
Get 了，我稍后去另一个 PR 改下。

Reply:
done https://github.com/cosname/cosx.org/pull/1019/commits/f0072df25e533783619072bd21d65508df5739a5

Reply:
还有个问题， https://cosx.org/2021/12/introduction-to-echarts4r 文章已经发布了，JS 的路径可以用正式路径吗？另外， Netlify 给的预览链接过段时间会失效吗？

Reply:
正式路径是指啥？

预览链接不知道会不会永久有效，至少我看了一下本站盘古开天辟地的第一个 PR #1 其预览链接依然有效，所以我估计应该大概率是永久有效的。

Reply:
> 正式路径是指啥？

文章发出后，查看网页源码，看到 JS 路径里没有 netlify 字样了。而是下面这个：

<img width="1082" alt="截屏2022-05-05 12 29 53" src="https://user-images.githubusercontent.com/12031874/166863789-823496d7-1029-4ced-ac14-1b5a5ecf94cb.png">

Reply:
那是浏览器的缓存吧。强制刷新一下（Cmd + Shift + R）。我看源码已经是 netlify.app 地址了。

Reply:
> 那是浏览器的缓存吧。

我后来新开了分支，获得预览链接，替换的。

就替换这几个 JS 文件，文章里的图都正常的，由此看来是引入了大量没有用的 JS，很奇怪 echarts4r 为啥这么做呢？

Reply:
不一定没用，可能会动态调用，比如语言文件，当然那一大堆国际语言文件中，大概也只会有一个有用，其它的都没用。

Reply:
是的。JS 文件可以像包一样被别的 JS 文件加载。